src/analytics.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import fs from 'fs-extra'
+import { default as OS } from 'os'
+// flow$ignore
+import type Config from 'cli-engine-config'
+import HTTP from 'cli-engine-command/lib/http'
+import Netrc from 'netrc-parser'
+import Output from 'cli-engine-command/lib/output'
+import URI from 'url'
+
+export default class AnalyticsCommand {
+  command: string
+  plugin: string
+  pluginVersion: string
+  version: string
+  platform: string
+  arch: string
+  start: any
+  config: Config
+
+  constructor (command: string, plugin: string, pluginVersion: string, config: Config) {
+    this.config = config
+    this.command = command
+    this.plugin = plugin
+    this.pluginVersion = pluginVersion
+    this.platform = config.platform
+    this.arch = OS.arch()
+    this.version = this.config.version
+  }
+
+  async record () {
+    if (AnalyticsCommand.skipAnalytics() || process.argv.length < 2) {
```

Comment:
this skips recording if either the analytics need to be skipped or there is no command. This could probably be done inside of cli.js for v6